waspc/examples/todoApp/ext/Todo.js

```diff
@@ -1,68 +1,92 @@
-// As seen here, we can import npm packages used in code generated by Wasp, which will be clearly defined.
-// In the future, we will of course also be able to specify additional packages as dependencies.
-import React from 'react'
-import { connect } from 'react-redux'
-
-// As seen here, we can import specific components/code generated by Wasp.
-// These will have well defined and documented APIs and paths.
-// Note that Task, NewTaskForm and TaskList are generated based on the declarations
-// we made in todoApp.wasp file.
-import NewTaskForm from '@wasp/entities/task/components/NewTaskForm'
-import TaskList from '@wasp/entities/task/components/TaskList'
-import * as taskState from '@wasp/entities/task/state.js'
-import * as taskActions from '@wasp/entities/task/actions.js'
-import ToggleIsDoneButton from '@wasp/components/ToggleIsDoneButton'
-import DeleteDoneButton from '@wasp/components/DeleteDoneButton'
-
-class Todo extends React.Component {
-  // TODO: prop types.
-
-  isAnyTaskCompleted = () => this.props.taskList.some(t => t.isDone)
-
-  isThereAnyTask = () => this.props.taskList.length > 0
-
-  render = () => {
-    return (
-      <div className="todos">
-        <div className="todos__container">
-          <h1> Todos </h1>
-
-          <div className="todos__toggleAndInput">
-            <ToggleIsDoneButton
-              disabled={!this.isThereAnyTask()}
-              className="todos__toggleButton"
-            />
+import React, { useState } from 'react'
+
+import { useQuery } from '@wasp/queries'
+import getTasks from '@wasp/queries/getTasks.js'
+import createTask from '@wasp/queries/createTask.js'
+
+const Todo = (props) => {
+  const [newTaskDescription, setNewTaskDescription] = useState('')
+
+  const { data: tasks, refetch, isFetching, isError, error: tasksError } = useQuery(getTasks)
+
+  const isAnyTaskCompleted = () => tasks?.some(t => t.isDone)
+
+  const isThereAnyTask = () => tasks?.length > 0
+
+  const createNewTask = async (description) => {
+    const task = { isDone: false, description }
+    await createTask(task)
+    refetch()
+  }
+
+  const handleNewTaskSubmit = async (event) => {
+    event.preventDefault()
+    try {
+      await createNewTask(newTaskDescription)
+      setNewTaskDescription('')
```

Comment:
`''` could be extracted into some variable as e.g. `taskDefaultDescription` or sth like that, now it is being duplicated (here and in `useState` in line 8)

Reply:
Very good point! I named it defaultNewTaskDescription.